### ISSUE TITLE
Invalid POD settings.jsonc files are not trapped

### ISSUE THREAD

**Issue by jkrasting** [NOAA-GFDL/MDTF-diagnostics#79]
In #73, there is an issue with an invalid `diagnostics/some_new_pod/settings.jsonc` file.  Syntax errors are not getting trapped by the framework on the POD discovery/import sequence.

This manifests itself as the new POD not showing up in `pod_list` and an assertion error:

```
Traceback (most recent call last):
  File "/Users/krasting/miniconda3/envs/_MDTF_base/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/Users/krasting/miniconda3/envs/_MDTF_base/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/krasting/mdtf/MDTF-diagnostics/src/mdtf.py", line 326, in <module>
    mdtf.main_loop()
  File "/Users/krasting/mdtf/MDTF-diagnostics/src/mdtf.py", line 291, in main_loop
    print(pod)
UnboundLocalError: local variable 'pod' referenced before assignment
```

I think we need to:

- flag the invalid jsonc file sooner
- document the POD discovery / call tree stack

 As a side note - looks like every online JSON validator I found online complains about comments in these `settings.jsonc` files.  Are there some tools we can point the POD develops to in order to check the validity of their files>

**Comment by tsjackson-noaa** [comment]
Hi @jkrasting, thanks for raising this as a separate issue, since it clarified where the bug was. NOAA-GFDL/MDTF-diagnostics#80 fixes the "flag the invalid jsonc file sooner" issue. "document the POD discovery / call tree stack" will be addressed by the changes I'm working on currently.

**Comment by tsjackson-noaa** [comment]
With respect to the online JSON validator issue, the online validators are technically correct: the JSON standard, by design, does not support comments ([wikipedia](https://en.wikipedia.org/wiki/JSON#As_a_configuration_language), see also [stackoverflow discussion](https://stackoverflow.com/questions/244777/can-comments-be-used-in-json)). 

What the framework currently uses are JSON files with C-style comments delimited by `//`. The comments are stripped off before the string is passed to python's JSON parser. I've been referring to this as ".jsonc" because that's the extension recognized by [VSCode](https://code.visualstudio.com/docs/languages/json#_json-with-comments), [github](https://github.com/github/linguist/issues/4153) and other text editors, but many not-really-official extensions to JSON have the same syntax, including "JSON5" (which isn't the "5th version of JSON" and isn't related to HTML5). 

The following online validators for these extensions to JSON *do* ignore the comments and check the syntax of the config files correctly: [jsonformatter.org/json5-validator](https://jsonformatter.org/json5-validator), [codebeautify.org/json5-validator](https://codebeautify.org/json5-validator). We should include these links in the docs.

**Comment by tsjackson-noaa** [comment]
Agree that the friction that using a quasi-standard format is causing for users tips the scales back in favor of using a simpler, more lightweight format -- YAML -- for config files. Migration to YAML has been moved to its own issue (MDTF-diagnostics#81).

**Comment by tsjackson-noaa** [comment]
Hi @jkrasting, OK with you if we close this out? The short-term fix (more careful parsing of comments) has been merged in as #80, and the long-term fix (YAML config files, rather than implementing our own comment parsing) has been given its own issue (#81).
